src/molecules/date-picker/useDatePicker.ts

```diff
@@ -0,0 +1,45 @@
+import { useState } from 'react'
+import { formatDateProperty, PropertyType } from '../../utils'
+
+const useDatePicker = ({ value, propertyType, disabled, onChange }:
+{
+  onChange: (date:string) => void,
+  value?:string|Date,
+  propertyType?:PropertyType,
+  disabled?:boolean
+}): {
+  dateString?: string,
+  date?: Date,
+  isCalendarVisible: boolean,
+  setCalendarVisible: (isVisible: boolean) => void,
+  onDateChange: (newDate: Date) => void
+} => {
+  const [isCalendarVisible, setCalendarVisible] = useState(false)
+
+  let date: Date | undefined
+  let dateString: string | undefined
+
+  if (value && value.constructor.name !== 'Date') {
+    // Safari doesn't parse value in format YYYY-MM-DD.
+    // All browsers are handling YYYY/MM/DD properly.
```

Comment:
actual fix

Reply:
Why do you use `value.constructor.name !== 'Date'` instead of `!(value instanceOf Date)`?